addons/udes_stock/models/stock_picking.py

```diff
@@ -3093,3 +3098,43 @@ def open_back_orders(self):
         return self.open_related_pickings(
             self.u_created_back_orders.ids, "Created Back Orders"
         )
+
+    @api.multi
+    def safe_prevent_date_done_on_not_done_pickings(self):
+        """
+        Set date_done field to empty in case that not all moves are in state done
+        """
+        for picking, moves in self.mapped("move_lines").groupby("picking_id"):
+            if picking.date_done and moves and not all(
+                    move.state in ("done", "cancel") for move in moves
+            ):
+                # Traceback into an issue to be able to trace more why happened in
+                # order to replicate it and solve the issue
+                picking.traceback_date_done_issue()
+                # Set date_done field to empty
+                picking.write({"date_done": False})
+
+    def traceback_date_done_issue(self):
+        """
+        Check if a project with a specific name exists
+        In case that exists create an issue with the traceback on it, otherwise first create
+        the project, after create issue for the created project.
+        """
+        # Give sudo access rights not to have access rights issue
+        self = self.sudo()
+        Project = self.env["project.project"]
+        Issue = self.env["project.task"]
+        self.ensure_one()
+        traceback_message = traceback.extract_stack()
+        prevent_name = "Prevent non-done pickings having date done set"
```

Comment:
This is a bit long for the project name, which we usually overview with kanban cards. But tricky to think of a title for this... maybe `Open Pickings with Date Done`?